### PR TITLE
Update to Swift 3

### PR DESCRIPTION
**Pull request by soffes**
⚠️ The two upload tests are currently failing so I commented them out. We should fix those before merging this.

### CODE REVIEW

**Comment by rhysforyou** [comment]
@soffes looks like you need to remove the `TOOLCHAINS` environment variable from _.travis.yml_, otherwise it's trying to build and test against Swift 2.3.

File: DVR/HTTPURLResponse.swift [hunk 1]
```diff
@@ -1,18 +1,18 @@
 import Foundation
 
-// There isn't a mutable NSHTTPURLResponse, so we have to make our own.
-class URLHTTPResponse: NSHTTPURLResponse {
+// There isn't a mutable HTTPURLResponse, so we have to make our own.
+class HTTPURLResponse: Foundation.HTTPURLResponse {
```
Comment (sethhoward) [inline DVR/HTTPURLResponse.swift:4]: If this code is not brought in via a pod (directly added to a project) which flavor of HTTPURLResponse will be used throughout the code base, Foundation or this? I would argue that this class should be prefixed.

**Comment by armstrongnate** [comment]
I fixed the upload tests in https://github.com/soffes/DVR/pull/1.

**Comment by codeOfRobin** [comment]
@soffes @n8armstrong Any ETA on when this'll be ready for production?

File: DVR/HTTPURLResponse.swift (hunk 1 above)
Comment (eliperkins) [reply DVR/HTTPURLResponse.swift:4]: You can directly reference via the module, that is, either `DVR.HTTPURLResponse` or `Foundation.HTTPURLResponse`.

I think having it that way is most explicit and readable.

**Comment by eliperkins** [comment]
@soffes @hyperspacemark @florin-pop @n8armstrong how do y'all feel about getting this PR up to date and merged into master? I feel like people have been using the `swift-3` branches/forks for long enough to vet them. It'd be great to get things like @n8armstrong's fixing of the upload tests and everyone else's work into master soon!

**Comment by brentleyjones** [comment]
Would love to have this merged!

**Pull request closed without merging**